lib/ClangImporter/ClangImporter.cpp

```diff
@@ -6877,6 +6877,22 @@ const clang::TypedefType *ClangImporter::getTypeDefForCXXCFOptionsDefinition(
   if (!enumDecl->getDeclName().isEmpty())
     return nullptr;
 
+  if (const clang::ElaboratedType *elaboratedType =
+          dyn_cast<clang::ElaboratedType>(
+              enumDecl->getIntegerType().getTypePtr())) {
+    if (auto typedefType =
+            dyn_cast<clang::TypedefType>(elaboratedType->desugar()))
+      if (auto enumExtensibilityAttr =
+              enumDecl->getAttr<clang::EnumExtensibilityAttr>();
+          enumExtensibilityAttr &&
+          enumExtensibilityAttr->getExtensibility() ==
+              clang::EnumExtensibilityAttr::Open &&
+          enumDecl->hasAttr<clang::FlagEnumAttr>()) {
+        return Impl.isUnavailableInSwift(typedefType->getDecl()) ? typedefType
+                                                                 : nullptr;
+      }
+  }
+
   if (auto typedefType = dyn_cast<clang::TypedefType>(
           enumDecl->getIntegerType().getTypePtr())) {
```

Comment:
pseudo code suggestion

```suggestion
  auto *t = enumDecl->getIntegerType().getTypePtr()
  if (const clang::ElaboratedType *elaboratedType =
          dyn_cast<clang::ElaboratedType>(t)) {
        t = elaboratedType->desugar();
  }

  if (auto typedefType = dyn_cast<clang::TypedefType>(t)) {
```

Reply:
@zoecarver I will try this out, but in the ElaboratedType case I did find that the check for the EnumExtensibilityAttr has to be on the enumDecl and not on typedefType->getDecl(). That's why I did it as a whole separate block just to keep things clear and to also keep them from interfering with the case that Nuri already addressed. 

Thoughts on this?

Reply:
I cleaned things up a bit and accounted for the different types for the attrs (EnumDecl vs typedefNamedDecl).